src/test/java/com/haxwell/apps/quizki/services/QuestionServiceImplTest.java

```diff
@@ -274,5 +335,78 @@ public void getQuestionByIdTest() {
 		assertThat(outputDTO.getId(), equalTo(1l));
 		
 	}
+	
+	@Test
+	public void getQuestionsTest() {
+		
+		
+		/*
+		 * Stated requirements
+		 * URL: /api/question?page=1&size=10
+		 * URL: /api/question?page=1 (size defaults to 10)
+		 * URL: /api/question?size=10 (page defaults to 1)
+		 * URL: /api/question (page and size are default)
+		 * 
+		 * parameters are passed to getQuestions() as received by the controller with defaults
+		 */
+		
+		int SEVENTEEN = 17;
+		questions = getQuestions(SEVENTEEN);
+		
+		when(questionRepo.count()).thenReturn((long)SEVENTEEN);
+		
+		Page<Question> pageOne = mock(Page.class);
+		PageRequest pageableOne = PageRequest.of(0,10);
+
+		when(questionRepo.findAll(pageableOne)).thenReturn(pageOne);
+		when(pageOne.getContent()).thenReturn(questions.subList(0, 10));
```

Comment:
I see how you are using the mock Page here. The code is easier to follow. The effect of mocking the page like this is to limit the responses to these fixed parameters in the method calls. So for example if I wanted to write a test for a page size of 2 or 8 I would need to write the mock responses for each one. The previous version would build a page per the parameters as would the actual repository. So is it better to mock the functionality or to create the functionality in the test?? I see the mock is easier to read but the functional is more flexible and would save time when more tests are added so I'm not sure what is more common or expected?

Reply:
There is a 2nd comment on the Controller file regarding the Pageable parameter passed to the Service. Not sure why it's not showing up here. Please check it out.

Reply:
You would probably (instead of all three cases in one test) break this down into individual tests, and use a common method to generate the mock Page and PageRequest. You could pass parameters into that method, and be able to test the various sizes. But you should remember, too, we're not testing Page and PageRequest, we're testing how this method uses them. So if we pass in a page size of 8, 100, or 32, it shouldn't matter. We expect that Page and PageRequest will do their job, if our method uses them correctly.